content/blog/2022-05-19-fast-brili.md

```diff
@@ -0,0 +1,141 @@
++++
+title = "Bril bytecode interpreter"
+[extra]
+bio = """
+	Charles Sherk is an junior studying Computer Science & Mathematics at
+	Cornell university, interested in compilers and programming language
+	implementation.
+	Susan Garry is a senior studying Computer Science at Cornell University,
+	interested in compilers and programming language design.
+
+"""
+[[extra.authors]]
+name = "Charles Sherk"
+link = "https://github.com/charles-rs/"
+[[extra.authors]]
+name = "Susan Garry"
+link = "https://github.com/susan-garry"
++++
+
+## Motivation
+Currently we have access to `brili`, which is written with an emphasis on
+readable code and extensiblity, and `brili-rs`, which is supposed to be faster,
+but `brili-rs` isn't actually that fast, and seems to be rather bloated for what
+it is. There were some points in this course where time was spent on "waiting
+for the Bril interpreter" that could have been avoided. So, we propose and
+implement an interpreter for Bril whose sole focus is speed.
+
+## Design overview
+
+### Language choice
+We chose C for an implementation language as it is known for its speed as well
+as simplicity, which means we won't inadvertently write code that is more
+expensive than necessary.
```

Comment:
Fine, but not sure how relevant this is. If you imagine yourself as a generally compiler-curious person stumbling across this blog posts, will they care why you chose C? I think they will care more about the virtues of a bytecode format for eliminating the overhead of AST interpretation.